### ISSUE TITLE
files structure

### ISSUE THREAD

**Issue by turican0** [GrimSqueaker/remc2#34]
In the future, it will be desirable to split sub_main.cpp / .h into multiple folders. I would wait to change these actions. But I would divide the part as soon as the current branches are merged. These are sounds and animation(videos).

I would create Basics.cpp/.h - some basic function used in both sound and animations only temporary solutions - may not be needed
I would create Sound.cpp/.h(include Basics)
I would create Animation.cpp/.h(include Basics, Sound)
I would create LevelGeneration.cpp/.h(include Basics)
I would change sub_main.cpp/.h(include Basics, Sound, Animations, LevelGeneration)

I would leave further divisions for later, because they are not unambiguous.

Independent LevelGeneration is good basis for editor - I want change it to project solution, but with use some files from remc2 project. Editor will be good for testing individual entities separately - for easy find some errors.

Do you agree with that?

Tom.

**Comment by GrimSqueaker** [comment]
Yes, this totally makes sense. Just do it for the Windows side and I can fix the CMake setup for Linux. But if you feel confident, you can also do the change in the CMakeLists.txt file to apply the change to Linux.

**Comment by turican0** [comment]
I started by dividing the functions into separate Sound and Animation files. This is only begin many functions is in bad files, now. The Basic file swelled a lot :) It will later need to be divided into things about rendering (Bitmap or Draw2D), things about memory (Memory) and more, but until then it would like to understand more what it is for. So far, I've left the files in remc2, so CMakeLists.txt probably wouldn't be affected.
Tom.

**Comment by turican0** [comment]
I begin with separate editor to own solution. Now is some code unclear, data is must copy to editor directory.
![image](https://user-images.githubusercontent.com/41465132/107414858-df028600-6b12-11eb-8d99-a3530913adff.png)
![image](https://user-images.githubusercontent.com/41465132/107414994-15d89c00-6b13-11eb-8c6f-e041c0c54a67.png)

**Comment by GrimSqueaker** [comment]
Ok, I have not tried to get the editor running on Linux. I think I would need to get kiss_sdl running for that.

**Comment by turican0** [comment]
Kiss-SDL is only 5 files, where is included in project(I modify original code adding some enhanced elements).

**Comment by turican0** [comment]
I'm thinking about further dividing functions into more files, so far this occurs to me, but in the future we should divide the functions even more into a more structured hierarchy in order to improve the orientation in the code.
![image](https://user-images.githubusercontent.com/41465132/108667536-b6e33180-74d9-11eb-9585-e32afe666654.png)
